OracleAnalytics/README.md

```diff
@@ -0,0 +1,279 @@
+to any future Oracle Analytics Server patch set release.
+
+### Starting a New Container
+
+The following variables are mandatory when starting the container:
+
+* ADMIN_USERNAME - WebLogic admin username for the new domain
+* ADMIN_PASSWORD - WebLogic admin password
+* DB_HOST - Host name for database into which to create schemas
+* DB_PORT - Database listener port
+* DB_SERVICE - Database instance service name
+* DB_USERNAME - Database sysdba username
+* DB_PASSWORD - Database sysdba password
+* SCHEMA_PREFIX - Schema prefix for new schemas in the new domain.
+The schema prefix must be unique. If it’s already in use, domain creation fails.
+* SCHEMA_PASSWORD - Password for all new schemas created
+
+ORACLE_HOME is fixed to `/u01/oracle`.
+DOMAIN_HOME is fixed to `/u01/oracle/user_projects/domains/bi`.
+
+For example:
+
+```bash
+docker run -d --name bi -p 9500:9500 -p 9502:9502 -e ADMIN_USERNAME=weblogic -e ADMIN_PASSWORD=<admin_password> -e DB_HOST=database -e DB_PORT=1521 -e DB_SERVICE=ORCLPDB1 -e DB_USERNAME=sys -e DB_PASSWORD=<db_password> -e SCHEMA_PREFIX=DEV -e SCHEMA_PASSWORD=<schema_password> oracle/analyticsserver:6.4-patch
+```
+
+Change _<...password>_ to your required values, and DB values to match your database.
+
+On success, logs the following:
+`The configuration of Oracle Fusion Middleware completed successfully.`.
+The script then tails the domain log to stdout.
+
+The BI domain uses the following ports:
+
+* 9500 - WebLogic AdminServer
+* 9502 - WebLogic bi_server1 managed server, hosting BI JEE apps
+* 9508 - OBI Cluster Controller, providing access to OBI Server system component
+* 9514 - OBI Server system component
+
+The following example URLs provide access to the BI domain:
+
+* <http://www.example.com:9500/console> - WebLogic Admin Console
+* <http://www.example.com:9502/analytics> - BI Answers/Dashboards
+* <http://www.example.com:9502/va> - Data Visualization
+* <http://www.example.com:9502/xmlpserver> - BI Publisher
+
+If access to the OBI Server component is required,
+for example by the BI Administration Tool,
+you must modify the previous `docker run` command to expose port 9514.
+
+For example:
+
+```bash
+docker run -it -p 9500:9500 -p 9502:9502 -p 9514:9514 -e ADMIN_USERNAME=weblogic -e ADMIN_PASSWORD=<admin_password> -e DB_HOST=database -e DB_PORT=1521 -e DB_SERVICE=ORCLPDB1 -e DB_USERNAME=sys -e DB_PASSWORD=<db_password> -e SCHEMA_PREFIX=DEV -e SCHEMA_PASSWORD=<schema_password> oracle/analyticsserver:6.4-patch
+```
+
+### Using a Host Directory for Persistent Data
+
+In the basic usage above,
+the container stores the persistent data in
+DOMAIN_HOME, APPLICATIONS_DIR, and SINGLETON_DATA_DIRECTORY.
+If the container is deleted, the data is lost.
+
+You can use a data volume for `/u01/oracle/user_projects`
+to store persistent data on the host disk.
+For example, to mount a host directory as a data volume:
+
+```bash
+mkdir -p /scratch/bi && chmod 777 /scratch/bi
```

Comment:
Never use `chmod 777` as this is insecure. You should be using `700` and ensuring that the user inside the container runs with the same UID/GID as the user that owns the files on the filesystem.

Reply:
Fixed